Defaultswitches

iandoug opened the pull request:
Dunno if you are waiting for this or working on it ...

ijprest commented:
Thanks... merged & published.

A few tweaks:
* I renamed the fields, mostly to make them shorter... :)
* I changed the data slightly to make it a bit more descriptive
  * e.g., instead of a simple string "MX Blue": "MX Blue, MX1A-E1xx, Normal, Tactile Clicky, 50 cN", I went with { "part": "MX1A-E1xx", "name": "MX Blue", "feel": "clicky", "weight": 50 }.  
  * Not much difference right now, but maybe in the future we could sort by weight, or something.
* In the metadata, I go with "IDs" for each style/brand/switch, and show a more descriptive name in the UI.
  * e.g., "cherry" --> "Cherry Electrical Products"
* I found part numbers for a few of the missing ones.
* I made it so when you change the mount-style, I "clear" the brand/switch-type, etc.

I updated PKB to match the changes.

iandoug commented:
hi Ian

On Thursday 06 August 2015 20:05:54 Ian Prest wrote:

> A few tweaks:

(comment suppressed .. :-) )

> * I changed the data slightly to make it a bit more descriptive
>   * e.g., instead of a simple string "MX Blue": "MX Blue, MX1A-E1xx, Normal,
> Tactile Clicky, 50 cN", I went with { "part": "MX1A-E1xx", "name": "MX
> Blue", "feel": "clicky", "weight": 50 }.

Good idea. Only Cherry makes (and/or specifies) the designed function of the 
switch (ie spacebar or normal), I guess we should assume the clones are all 
meant for 'normal'.

>* In the
> metadata, I go with "IDs" for each style/brand/switch, and show a more
> descriptive name in the UI. * e.g., "cherry" --> "Cherry Electrical
> Products"

Okay, then this:
    "gaote": { "brand": "gaote", "name": "Outemu / Gaote Electronics", 
"switches": [

Should be 

    "Outemu": { "brand": "Outemu", "name": "Gaote Electronics", "switches": [

It's like with Kailh .. brand name is different to company name.

When I was originally doing the json, I had issues using the "/" so I gave up 
struggling and worked around it with the "_".


> * I found part numbers for a few of the missing ones.

Ah, I didn't think of doing what you did with Kailh to cover the different 
mount styles. I was going to go with xx like with Cherry, because I was 
thinking to add another field to that page:

* Switch mounting: []PCB []Plate

And somewhere down the line, to use what's selected there to pick the 
appropriate part number. In truth, you probably need more options, like
* Diode
* LED 
as well, at which point it gets a bit fiddly... :-)

> * I made it so when you change the mount-style, I "clear" the
> brand/switch-type, etc.

thanks, will examine your code to learn the trick ;-)

> 
> I updated PKB to match the changes.

Thanks...

What are you working on next?

cheers, Ian


-- 
ian@zti.co.za http://www.zti.co.za
Zero 2 Infinity - The net.works
Phone +27-21-975-7273

iandoug commented:
On Friday 07 August 2015 08:21:03 Ian Douglas wrote:

> Okay, then this:
>     "gaote": { "brand": "gaote", "name": "Outemu / Gaote Electronics", 
> 
> Should be 
> 
>     "Outemu": { "brand": "Outemu", "name": "Gaote Electronics", 

Apologies, should be 
"outemu": { "brand": "outemu", "name": "Outemu (Gaote Electronics)", 

I fixed it in my master branch and also since you were one jump ahead of me 
with the WASD colours, just fixed the colour count and URL, since the one in 
the json file no longer has the colours on...

cheers, Ian

-- 
ian@zti.co.za http://www.zti.co.za
Zero 2 Infinity - The net.works
Phone +27-21-975-7273
